internal/kafka/internal/workers/kafka_mgrs/promotion/internal/actions/delete_actual_quota_action.go

```diff
@@ -0,0 +1,53 @@
+package actions
+
+import (
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/api/dbapi"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/config"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/services"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/workers/kafka_mgrs/promotion/chain"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/api"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/errors"
+)
+
+var _ chain.ReconcileAction[PromotionContext] = &DeleteActualQuotaAction{}
+
+type DeleteActualQuotaAction struct {
+	quotaServiceFactory services.QuotaServiceFactory
+	kafkaConfig         config.KafkaConfig
+}
+
+func NewDeleteActualQuotaAction(kafkaConfig config.KafkaConfig, quotaServiceFactory services.QuotaServiceFactory) chain.ReconcileAction[PromotionContext] {
+	return &DeleteActualQuotaAction{
+		quotaServiceFactory: quotaServiceFactory,
+		kafkaConfig:         kafkaConfig,
+	}
+}
+
+func (d *DeleteActualQuotaAction) PerformJob(kafkaRequest *dbapi.KafkaRequest, currentResult chain.Result[PromotionContext]) (chain.Result[PromotionContext], bool, error) {
+	res := chain.Result[PromotionContext]{}
+	quotaService, factoryErr := d.quotaServiceFactory.GetQuotaService(api.QuotaType(d.kafkaConfig.Quota.Type))
+	if factoryErr != nil {
+		return res, true, errors.NewWithCause(errors.ErrorGeneral, factoryErr, "unable to delete quota")
+	}
+	instanceType, err := d.kafkaConfig.SupportedInstanceTypes.Configuration.GetKafkaInstanceTypeByID(kafkaRequest.InstanceType)
+	if err != nil {
+		// instance type was validated at creation stage. This should never happen.
+		return res, true, err
+	}
+	bm, err := instanceType.GetKafkaSupportedBillingModelByID(kafkaRequest.ActualKafkaBillingModel)
+	if err != nil {
+		// actual billing model was validated at creation stage. This should never happen.
+		return res, true, err
+	}
+
+	// WARNING: all the other methods called before are returning an `error`. Quota service methods
```

Comment:
In this cases I find helpful to explicitely assign a non standard name to the error. For example `svcErr`